### PR TITLE
add the groups resources service addition

### PR DESCRIPTION
**Pull request by ryanfkeepers**
Adds groups to the m365 services api.  Also adds a bit of touchups/cleanups on the side.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature

#### Issue(s)

* #3989

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/pkg/services/m365/groups.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	}
+
+	return ret, nil
+}
+
+// ---------------------------------------------------------------------------
+// helpers
+// ---------------------------------------------------------------------------
+
+// parseUser extracts information from `models.Userable` we care about
```
Comment (meain) [inline src/pkg/services/m365/groups.go:83]:
```suggestion
// parseUser extracts information from `models.Groupable` we care about
```

File: src/pkg/services/m365/api/helper_test.go [hunk 2]
```diff
@@ -132,13 +132,14 @@ func newIntegrationTesterSetup(t *testing.T) intgTesterSetup {
 
 	its.siteDriveRootFolderID = ptr.Val(siteDriveRootFolder.GetId())
 
-	// teams
-	its.teamID = tconfig.M365TeamsID(t)
+	// group
 
-	team, err := its.ac.Groups().GetTeamByID(ctx, its.teamID)
+	its.groupID = tconfig.M365TeamID(t)
```
Comment (meain) [inline src/pkg/services/m365/api/helper_test.go:137]:
```suggestion
	its.groupID = tconfig.M365GroupID(t)
```

**Review by meain — APPROVED** [review]
Few small nites, but LGTM.

File: src/pkg/services/m365/api/helper_test.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/pkg/services/m365/api/helper_test.go:137]: That's actually intentional.  I've added a comment to explain the choice.  In short: we want to ensure the group supports teams apis for full testing coverage.

**Pull request merged**